remote_shell_service.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"bufio"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/subtle"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"golang.org/x/term"
+)
+
+type Console struct {
+	stdInLock  sync.Mutex
+	stdInPipe  io.Writer
+	stdOutPipe io.Reader
+	stdErrPipe io.Reader
+
+	sessionLock    sync.Mutex
+	remoteSessions map[uuid.UUID]ssh.Session
+}
+
+func makeConsole(stdin io.Writer, stdout io.Reader, stderr io.Reader) Console {
+	return Console{
```

Comment:
Would be slightly better to avoid a copy of the thing just created:

```suggestion
func makeConsole(stdin io.Writer, stdout io.Reader, stderr io.Reader) *Console {
	return &Console{
```